torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -143,55 +220,47 @@ void compileKernel(Fusion& fusion, CudaKernel& entry) {
   ptx.resize(ptx_size);
   AT_CUDA_NVRTC_CHECK(nvrtc().nvrtcGetPTX(program, ptx.data()));
 
-  AT_CUDA_DRIVER_CHECK(nvrtc().cuModuleLoadData(&(entry.module_), ptx.data()));
+  AT_CUDA_DRIVER_CHECK(nvrtc().cuModuleLoadData(&(entry->module_), ptx.data()));
   AT_CUDA_DRIVER_CHECK(nvrtc().cuModuleGetFunction(
-      &(entry.function_), entry.module_, lowered_kernel_name));
+      &(entry->function_), entry->module_, lowered_kernel_name));
   AT_CUDA_DRIVER_CHECK(nvrtc().cuOccupancyMaxActiveBlocksPerMultiprocessor(
-      &entry.max_blocks_, entry.function_, 128, 0));
-  entry.max_blocks_ *= prop->multiProcessorCount;
+      &entry->max_blocks_, entry->function_, 128, 0));
+  entry->max_blocks_ *= prop->multiProcessorCount;
 }
 
 void runKernel(
-    CudaKernel& entry,
+    CudaKernel* entry,
     const at::ArrayRef<IValue>& inputs,
     std::vector<at::Tensor>& outputs) {
   const auto prior_device = at::cuda::current_device();
-  at::cuda::set_device(entry.device_);
+  at::cuda::set_device(entry->device_);
   auto stream = at::cuda::getCurrentCUDAStream();
 
   // TODO: Proper API to establish reasonable launch configurations;
   // Naive launch config;
   size_t numel = outputs[0].numel();
-  const auto nBlocks = std::min(entry.max_blocks_, ceilDiv(numel, 128));
 
-  // TODO: Proper API to tranform JIT I/O Tensor to CodeGen I/O Tensor
-  std::vector<void*> arguments;
+  // TODO: we can't randomly clap down this until we got striding.
+  // const auto nBlocks = std::min(entry->max_blocks_, ceilDiv(numel, 128));
+  const auto nBlocks = ceilDiv(numel, 128);
 
-  // TODO: There are better ways to do this;
-  // argument holder;
-  // host code, `T` in `Tensor<T>` doesn't really matter, as we only interact
-  // with the address; Just put a float here to simply the argument holder.
+  // TODO: Proper API to tranform JIT I/O Tensor to CodeGen I/O Tensor
   auto max_capacity = inputs.size() + outputs.size();
-  std::vector<Tensor<float>> tensor_args;
-  std::vector<int> int_args;
-  std::vector<float> float_args;
-  tensor_args.reserve(max_capacity);
-  int_args.reserve(max_capacity);
-  float_args.reserve(max_capacity);
+  KernelArgumentHolder kernel_arg_holder(max_capacity, outputs[0].dim());
 
   // Naive I/O setup, I'm ignoring all the potential transformation (i.e. I/O
```

Comment:
We can delete this comment, as root TensorDomain does have matching dimension as the IO tensor.